File: Cpp/646_Maximum_Length_of_Pair_Chain.cpp
class Solution {
public:
    int findLongestChain(vector<vector<int>>& pairs) {
        int result = 1;
        sort(pairs.begin(), pairs.end(), [](vector<int> a, vector<int> b){
            if(a[1] == b[1]) return a[0] < b[0];
            return a[1] < b[1];
        });
        
        int current = pairs[0][1];
        for(auto& pair : pairs){
            if(current >= pair[0]) continue;
            current = pair[1];
            ++result;
        }
        
        return result;
    }
};

File: Kotlin/646_Maximum_Length_of_Pair_Chain.kt
class Solution {
    fun findLongestChain(pairs: Array<IntArray>): Int {
        var result = 0
        val n = pairs.size
        var current = Int.MIN_VALUE
        
        Arrays.sort(pairs){ a, b ->
            a[1] - b[1]
        }
        
        for(i in 0 until n){
            if(current >= pairs[i][0]) continue
            current = pairs[i][1]
            ++result
        }
        
        return result
    }
}